distributed/tests/test_utils_test.py

```diff
@@ -919,17 +920,13 @@ def test_popen_timeout(capsys):
                     import sys
                     import time
 
-                    if sys.platform == "win32":
-                        signal.signal(signal.SIGBREAK, signal.default_int_handler)
-                        # ^ Cause `CTRL_BREAK_EVENT` on Windows to raise `KeyboardInterrupt`
+                    signum = signal.SIGBREAK if sys.platform == "win32" else signal.SIGINT
+                    signal.signal(signum, signal.SIG_IGN)
+                    print("ready", flush=True)
 
-                    print('ready', flush=True)
                     while True:
-                        try:
-                            time.sleep(0.1)
-                            print("slept", flush=True)
-                        except KeyboardInterrupt:
-                            print("interrupted", flush=True)
```

Comment:
The issue was that the code would enter and exit the `try... except KeyboardInterrupt` block, so it was possible that the SIGINT would land outside of the block, cleanly terminating the subprocess.
Unsure why it seems to affect exclusively Linux.